tests/data.json

```diff
@@ -0,0 +1,226 @@
+                "name": "fake-async-plan",
+                "id": "fake-async-plan-guid",
+                "description": "Shared fake Server, 5tb persistent disk, 40 max concurrent connections. 100 async",
+                "max_storage_tb": 5,
+                "metadata": {
+                  "cost": 0,
+                  "bullets": [
+                    {
+                      "content": "40 concurrent connections"
+                    }
+                  ]
+                }
+              },
+              {
+                "name": "fake-async-plan-2",
+                "id": "fake-async-plan-2-guid",
+                "description": "Shared fake Server, 5tb persistent disk, 40 max concurrent connections. 100 async",
+                "max_storage_tb": 5,
+                "metadata": {
+                  "cost": 0,
+                  "bullets": [
+                    {
+                      "content": "40 concurrent connections"
+                    }
+                  ]
+                }
+              }
+            ]
+          }
+        ]
+      }
+    },
+    "provision": {
+      "fake-async-plan-guid": {
+        "sleep_seconds": 0,
+        "status": 202,
+        "body": {
+          "last_operation": {
+            "state": "in progress"
+          }
+        }
+      },
+      "fake-async-plan-2-guid": {
+        "sleep_seconds": 0,
+        "status": 202,
+        "body": {
+          "last_operation": {
+            "state": "in progress"
+          }
+        }
+      },
+      "default": {
+        "sleep_seconds": 0,
+        "status": 200,
+        "body": {}
+      }
+    },
+    "fetch": {
+      "default": {
+        "in_progress": {
+          "sleep_seconds": 0,
+          "status": 200,
+          "body": {
+            "last_operation": {
+              "state": "in progress"
+            }
+          }
+        },
+        "finished": {
+          "sleep_seconds": 0,
+          "status": 200,
+          "body": {
+            "last_operation": {
+              "state": "succeeded"
+            }
+          }
+        }
+      }
+    },
+    "update": {
+      "fake-async-plan-guid": {
+        "sleep_seconds": 0,
+        "status": 202,
+        "body": {
+          "last_operation": {
+            "state": "in progress"
+          }
+        }
+      },
+      "fake-async-plan-2-guid": {
+        "sleep_seconds": 0,
+        "status": 202,
+        "body": {
+          "last_operation": {
+            "state": "in progress"
+          }
+        }
+      },
+      "default": {
+        "sleep_seconds": 0,
+        "status": 200,
+        "body": {
+          "last_operation": {
+            "state": "succeeded"
+          }
+        }
+      }
+    },
+    "deprovision": {
+      "fake-async-plan-guid": {
+        "sleep_seconds": 0,
+        "status": 202,
+        "body": {
+          "last_operation": {
+            "state": "in progress"
+          }
+        }
+      },
+      "fake-async-plan-2-guid": {
```

Comment:
may be also cover the case of a service with async provisionning, and **sync** deprovisionning.

Reply:
Unfortunately this test broke. The way I managed to fix it to only allowing asynchronous deprovisioning.

Reply:
thanks. This is a corner case we can probably live with for now. You may record the limitation into a distinct issue ?

Reply:
Limitation has been added https://github.com/mevansam/terraform-provider-cf/issues/107